test/sql/alter/test_rename_table.cpp

```diff
@@ -0,0 +1,108 @@
+#include "catch.hpp"
+#include "test_helpers.hpp"
+
+using namespace duckdb;
+using namespace std;
+
```

Comment:
I would want to see the following extra tests:

(1) ALTER TABLE where the table does not exist (e.g. `ALTER TABLE nonexisting_table RENAME TO tbl2`)
(2) ALTER TABLE where destination table already exists (e.g. `ALTER TABLE tbl1 RENAME TO existing_tbl`)
(3) I'm also missing a storage test: see `test_store_alter.cpp` for an example of this. If we rename an existing table we want to make sure that the rename is correctly written to the WAL and is persisted to disk.

Reply:
I will add those tests.